File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/Administration/Announcement.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.Administration
{
    public class Announcement : AuditableEntity
    {
        [Key]
        public long ID { get; set; } 
        public string ImageLink { get; set; }
        public string Title { get; set; }

        public ICollection<AnnouncementSharedTo> AnnouncementSharedTos { get; set; } = new List<AnnouncementSharedTo>();
    }
}

File: LMS_CMS_Frontend/src/app/Models/Administrator/announcement.ts
import { AnnouncementSharedTo } from "./announcement-shared-to";

export class Announcement {
    constructor(
        public id: number = 0,
        public title: string = '',
        public imageLink: string = '',
        public insertedByUserId :number =0,
        public announcementSharedTos : AnnouncementSharedTo[]=[],
        public userTypeIDs : number[]=[],
        public imageFile : File|null = null
    ) {}
} 